env/mock_env.cc

```diff
@@ -780,4 +780,14 @@ void MockEnv::FakeSleepForMicroseconds(int64_t micros) {
   fake_sleep_micros_.fetch_add(micros);
 }
 
+#ifndef ROCKSDB_LITE
+// This is to maintain the bahavior before swithcing from InMemoryEnv to MockEnv
+Env* NewMemEnv(Env* base_env) { return new MockEnv(base_env); }
```

Comment:
can we add `ROCKSDB_DEPRECATED_FUNC` to its declaration and then we should be able to remove it after a few versions